pkg/ccl/importccl/read_import_pgdump.go

```diff
@@ -102,3 +108,212 @@ func (p *postgreStream) read() (tree.StatementList, error) {
 		}
 	}
 }
+
+func readPostgresCreateTable(
+	input io.Reader, evalCtx *tree.EvalContext, settings *cluster.Settings, match string, parentID sqlbase.ID, walltime int64,
+) ([]*sqlbase.TableDescriptor, error) {
+
+	// Modify the CreateTable stmt with the various index additions. We do this
+	// instead of creating a full table descriptor first and adding indexes
+	// later because MakeSimpleTableDescriptor calls the sql package which calls
+	// AllocateIDs which adds the hidden rowid and default primary key. This means
+	// we'd have to delete the index and row and modify the column family. This
+	// is much easier and probably safer too.
+	createTbl := make(map[string]*tree.CreateTable)
+	var found []string
+	ps := newPostgreStream(input, defaultPgStreamInit, defaultPgStreamMax)
+	for {
+		stmt, err := ps.Next()
+		if err == io.EOF {
+			if len(createTbl) == 0 {
+				return nil, errors.Errorf("no table definition found")
+			}
+			ret := make([]*sqlbase.TableDescriptor, 0, len(createTbl))
+			for name, create := range createTbl {
+				if match == "" || match == name {
```

Comment:
we're filtering based on `match` here but I think we'd never add to the map because we're also filtering below? do we need both? If not, it seems like it'd be clearer with only one.

2¢: I'd probably just have the map that has keys for every table name seen, with non-nil entries for those that matched. Then we can also skip updating `found`, and just make it in the error case from the map keys.

Reply:
done